src/Node/Express/Request.js

```diff
@@ -138,3 +138,15 @@ exports._getOriginalUrl = function (req) {
     };
 };
 
+exports._getData = function (req, field) {
+    return function () {
+        return req[field];
+    };
+};
+
+exports._setData = function (req, field, val) {
+    return function () {
+        req[field] = val;
```

Comment:
WDYT if instead of writing data straight to `req` object, we write it into some child object, e.g. `req.userData`? This way user won't occasionally override any of request's fields.